addons/mail/models/mail_scheduled_message.py

```diff
@@ -172,27 +172,43 @@ def _post_message(self, raise_exception=True):
-            the _post_messages_cron
+            instead of raising an error, and send a notification to the author about the failure.
+            This is useful when scheduled messages are sent from the _post_messages_cron.
         """
         notification_parameters_whitelist = self._notification_parameters_whitelist()
         for scheduled_message in self:
             message_creator = scheduled_message.create_uid
             try:
-                scheduled_message.with_user(message_creator)._check()
-            except AccessError:
+                with self.env.cr.savepoint():
+                    scheduled_message.with_user(message_creator)._check()
+                    self.env[scheduled_message.model].browse(scheduled_message.res_id).with_user(message_creator).message_post(
+                        attachment_ids=list(scheduled_message.attachment_ids.ids),
+                        author_id=scheduled_message.author_id.id,
+                        body=scheduled_message.body,
+                        partner_ids=list(scheduled_message.partner_ids.ids),
+                        subtype_xmlid='mail.mt_note' if scheduled_message.is_note else 'mail.mt_comment',
+                        **{k: v for k, v in json.loads(scheduled_message.notification_parameters or '{}').items() if k in notification_parameters_whitelist},
+                    )
+            except Exception:
                 if raise_exception:
                     raise
-                _logger.info("Posting of scheduled message %s failed: user %s cannot post on the record", scheduled_message.id, message_creator.id)
-                continue
-            self.env[scheduled_message.model].browse(scheduled_message.res_id).with_user(message_creator).message_post(
-                attachment_ids=list(scheduled_message.attachment_ids.ids),
-                author_id=scheduled_message.author_id.id,
-                body=scheduled_message.body,
-                partner_ids=list(scheduled_message.partner_ids.ids),
-                subtype_xmlid='mail.mt_note' if scheduled_message.is_note else 'mail.mt_comment',
-                **{k: v for k, v in json.loads(scheduled_message.notification_parameters or '{}').items() if k in notification_parameters_whitelist},
-            )
+                _logger.info("Posting of scheduled message with ID %s failed", scheduled_message.id, exc_info=True)
+                # notify user about the failure (send content as user might have lost access to the record)
+                try:
+                    with self.env.cr.savepoint():
+                        self.env['mail.thread'].message_notify(
+                            partner_ids=[message_creator.partner_id.id],
+                            subject=_("A scheduled message couldn't be sent"),
+                            body=_("The message scheduled on %(model)s(%(id)s) with the following content could not be sent: %(original_message)s",
```

Comment:
Shouldn't we format a bit, like when having a forwarded message ? like splitting with <br> at least, adding a --- original content line break, ...